src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -379,85 +488,116 @@ void Snippet::initOptimalPrimitiveDescriptor() {
+    uint8_t mask = 0;
+    // TODO: add check for non-eltwise inputs
+    for (const auto& shape : input_shapes) {
+        mask = mask << 1;
+        if (shape.back() == 1)
+            mask = mask | 1;
     }
-
-    return getMaxPrecision(inputPrecisions);
+    return mask;
 }
 
-void Snippet::prepareParams() {
-    for (size_t i = 0; i < inputNum; i++)
-        snippetAttrs.inMemBlockedDims[i] = getParentEdgeAt(i)->getMemory().getDescWithType<BlockedMemoryDesc>()->getBlockDims();
-    for (size_t i = 0; i < outputNum; i++)
-        snippetAttrs.outMemBlockedDims[i] = getChildEdgeAt(i)->getMemory().getDescWithType<BlockedMemoryDesc>()->getBlockDims();
+void Subgraph::lower() {
+    const auto& subgraph = subgraph_attrs->snippet;
 
-    SnippetKey key = {snippetAttrs};
+    const auto in_blocked_shapes = getSnippetsBlockedShapes();
+    const auto precisions = getIOPrecisions();
+    subgraph->data_flow_transformations(in_blocked_shapes, precisions.first, precisions.second, getDataFlowPasses());
 
-    auto builder = [this](const SnippetKey& key) -> std::shared_ptr<SnippetExecutor> {
-        std::shared_ptr<SnippetExecutor> executor =
-                std::make_shared<SnippetJitExecutor>(key.attrs, is_dynamic);
-        return executor;
-    };
+    // TODO: Snippets don't support backend-provided blocking, so we need to reshape body
+    //       using blocked shapes first. This can be removed after [121670]
+    std::vector<snippets::VectorDimsRef> in_shapes;
+    for (const auto& s : in_blocked_shapes)
+        in_shapes.emplace_back(s.first);
+    subgraph->shape_infer(in_shapes);
+
+#ifdef SNIPPETS_LIBXSMM_TPP
+    subgraph->set_tile_rank(std::min(2ul, subgraph->infer_master_shape().size()));
+#endif
+
+    const auto control_flow_settings = getControlFlowPasses();
+    // Note: minimal JIT work amount is a predefined value that describes the number of kernel iterations (work amount)
+    // needed to cover kernel call overhead. It is used for balancing between parallel and JIT work amounts in domain optimization.
+    subgraph->lowering_transformations(static_cast<size_t>(parallel_get_max_threads()), 256,
+                                       std::make_shared<snippets::CPUShapeInferSnippetsFactory>(),
+                                       control_flow_settings.first, control_flow_settings.second);
+}
 
-    auto getOrCreateExecutor = [this, &key, &builder]() {
-        auto cache = context->getParamsCache();
-        auto result = cache->getOrCreate(key, builder);
-        execPtr = result.first;
-        if (!execPtr) {
-            OPENVINO_THROW("Executor is not created for node ", getName(), ".");
+void Subgraph::prepareParams() {
+    const auto cache = context->getParamsCache();
+
+    auto builder = [this, cache](const SubgraphKey& key) -> std::shared_ptr<SubgraphExecutor> {
+        const auto& cpu_config = ov::as_type_ptr<CPURuntimeConfig>(subgraph_attrs->snippet->update_runtime_config());
```

Comment:
A bit confusing, `cpu_config` is more about things like isa
```suggestion
        const auto& snippet_config = ov::as_type_ptr<CPURuntimeConfig>(subgraph_attrs->snippet->update_runtime_config());
```